SpatialGDK/Source/SpatialGDKServices/Private/SpatialCommandUtils.cpp

```diff
@@ -496,4 +497,55 @@ bool SpatialCommandUtils::GetProcessInfoFromPort(int32 Port, FString& OutPid, FS
 	return false;
 }
 
+void SpatialCommandUtils::FetchRuntimeBinary(const FString& RuntimeVersion)
+{
+	FString RuntimePath = FPaths::Combine(SpatialGDKServicesConstants::GDKProgramPath, TEXT("runtime"), RuntimeVersion);
+	SpatialCommandUtils::FetchPackageBinary(RuntimeVersion, SpatialGDKServicesConstants::RuntimeExe, "runtime", RuntimePath, true);
+}
+
+void SpatialCommandUtils::FetchInspectorBinary(const FString& InspectorVersion)
+{
+	FString InspectorPath = FPaths::Combine(SpatialGDKServicesConstants::GDKProgramPath, TEXT("inspector"), InspectorVersion, SpatialGDKServicesConstants::InspectorExe);
+	SpatialCommandUtils::FetchPackageBinary(InspectorVersion, SpatialGDKServicesConstants::InspectorExe, "inspector", InspectorPath, false);
+}
+
+void SpatialCommandUtils::FetchPackageBinary(const FString& PackageVersion, const FString& PackageExe, const FString& PackageName, const FString& SaveLocation, const bool bUnzip)
+{
+	FString PackagePath = FPaths::Combine(SpatialGDKServicesConstants::GDKProgramPath, *PackageName, PackageVersion);
+
+	// Check if the binary already exists for a given version
+	if (FPaths::FileExists(FPaths::Combine(PackagePath, PackageExe)))
+	{
+		UE_LOG(LogSpatialCommandUtils, Verbose, TEXT("%s binary already exists."), *PackageName);
+		return;
+	}
+
+	// If it does not exist then fetch the binary using `spatial worker package retrieve`
+	UE_LOG(LogSpatialCommandUtils, Log, TEXT("Trying to fetch %s version %s"), *PackageName, *PackageVersion);
+	FString Params = FString::Printf(TEXT("package retrieve %s %s %s %s"), *PackageName, *SpatialGDKServicesConstants::PlatformVersion,
+									 *PackageVersion, *SaveLocation);
+	if (bUnzip)
+	{
+		Params += " --unzip";
+	}
+
+	TOptional<FMonitoredProcess> FetchingProcess;
+	const FString& ExePath = SpatialGDKServicesConstants::SpatialExe;
+	FetchingProcess = { ExePath, Params, true, true };
+	FetchingProcess->OnOutput().BindLambda([](const FString& Output) {
+		UE_LOG(LogSpatialCommandUtils, Display, TEXT("Runtime: %s"), *Output);
```

Comment:
```suggestion
		UE_LOG(LogSpatialCommandUtils, Display, TEXT("FetchingProcess: %s"), *Output);
```

Reply:
wait, this is just for logging, like it should be either Worker or Runtime, no?

Reply:
This is the output of the `spatial` process itself. You're not running the runtime here or any worker.